WooCommerce/Classes/ViewRelated/ReusableViews/SwiftUI Components/TextFieldStyles.swift

```diff
@@ -2,15 +2,27 @@ import SwiftUI
 
 /// Text field has a rounded border that has a thicker border and brighter border color when the field is focused.
 struct RoundedBorderTextFieldStyle: TextFieldStyle {
-    let focused: Bool
+    private let focused: Bool
+    private let insets: EdgeInsets
+
+    init(focused: Bool, insets: EdgeInsets = Defaults.insets) {
+        self.focused = focused
+        self.insets = insets
+    }
 
     func _body(configuration: TextField<Self._Label>) -> some View {
         configuration
-            .padding(10)
+            .padding(insets)
```

Comment:
Looks like the change of height when switching between field types is due to vertical padding and the content height difference of `SecureField` and `TextField`.

A quick fix would be to add a fixed height, we may need to take scale into consideration:

```suggestion
            .padding(insets)
            .frame(height: 44 * scale)
```

Reply:
After some testing, the insets aren't the cause of the mismatched height, the height is just slightly different between the `TextField` and `SecureField`. In https://github.com/woocommerce/woocommerce-ios/pull/8006/commits/a09e7e3911290142d76b25ab59b9ae9b8c499c09, I passed an optional `height` parameter for `AccountCreationFormFieldView` to set `44 * scale`.